### DISCUSSION TITLE
Enhance custom select support with all-fields-optional derive or into_model with defaults

### DISCUSSION THREAD

**Discussion by jharrilim** [SeaQL/sea-orm#1448] [Ideas]
## Motivation

I would like to use sea-orm alongside async-graphql. In GraphQL, one of the features it has is providing specific fields to select. This is helpful for reducing data transferred across the server to the client, but we can also use it to optimize our SQL queries. Deferring to the client which fields to select however comes with the drawback of all fields becoming optional. For example, this model may make sense to represent a table:

```rust
#[derive(Debug, Default, Clone, Eq, PartialEq, DeriveEntityModel, Serialize, Deserialize, SimpleObject)]
#[graphql(name = "User")]
#[sea_orm(table_name = "users")]
pub struct Model {
  id: i32,
  email: String,
  created_at: DateTime,
  deleted_at: Option<DateTime>,
}
```

But, when performing a select where all of the column choices are deferred to the client:

```rust
// pretend it is generated from GraphQL field selections
let columns: Vec<user::Column> = vec![user::Column::Email];

let select = columns.fold(User::find().select_only(), |sel, col| {
  sel.column(col)
});

select.all(&db)
```

The optionality of the original struct tends to not match.

## Proposed Solutions

Initially I was thinking to derive an all-fields-optional struct from the model:

```rust
#[derive(SimpleObject)]
pub struct OptionalModel {
  id: Option<i32>,
  email: Option<String>,
  created_at: Option<DateTime>,
  deleted_at: Option<DateTime>,
}
```

This might be somewhat tricky and I'm not too sure if it would entirely work. An easier way might be to add a method similar to `into_model`, but one that makes use of a default implementation to fill missing fields:

```rust
// pretend it is generated from user input
let columns: Vec<user::Column> = vec![user::Column::Email];

let select = columns.fold(User::find().select_only(), |sel, col| {
  sel.column(col)
});

select.into_model_with_defaults().all(&db)
```

It works well in the case of GraphQL since those defaulted fields aren't being returned anyway, and as long as it is opt-in, it shouldn't cause any confusion.

**Comment by tyt2y3** [comment]
The 'nullable' problem is already solved by `ActiveModel` where each field can be `unset`.
So the problem is adding `SelectOptional` that return `ActiveModel` instead of `Model`
